Mission 완료 | 🐸 리팩토링 진행중

orosy opened the pull request:
## 수정 사항 간략한 한줄 요약
- 초기 셋팅 수정 완료
- Mission 1) `Login` | 페이지 사용자 입력 데이터 저장 구현
- Mission 2) `Login` | 로그인 버튼 활성화 (validation) 구현
- Mission 3) `Main` | 댓글 기능 구현
- Mission 4) `Main` | 댓글 컴포넌트화 + props로 데이터 전달 구현
- Additional Features) `Main` | Feed 좋아요 토글 버튼 구현
- Additional Features) `Main` | 댓글 좋아요 토글 버튼 구현
- Additional Features) `Main` | 댓글 삭제 버튼 구현
- Additional Features) `Main` | 부모 페이지로 단방향 데이터 흐름 구현
 		 
## 수정 사항들 자세한 내용
- `Login`, `Main` 페이지 React.js로 변경 진행
- `Nav` 부분 팀 공통으로 진행하여 프로젝트에 적용
- `Aside` 유지보수 위한 컴포넌트화 완료
- 댓글 입력값 유무에 따른 `게시` 버튼색 변화 구현
- `Fontawesome icon` 클래스명 JSX화
- `Comment`에 있던 `state`, `event` 등 `Main`으로 데이터 이동
- `Comment`의 자식 컴포넌트에 `props`로 데이터 전달 구현
- `Login` 페이지 가독성 위한 구조분해 할당 진행
- `CommetOne` 컴포넌트 가독성 위한 프로퍼티 이름 변경
- `CommentList` 컴포넌트, 파일명 일치화 및 코드 리팩토링
- `FeedLike` 컴포넌트명 변경으로 인한 재사용 원활화
- `Feed` 내 데이터 모두 `Mock Data` 변경
- `Login` 페이지 유효성 검사 로직 리팩토링 
- `Mock Data` 사용 후, 달라진 context에 따른 댓글 추가 기능 리팩토링
- `fetch` 함수 사용하여 백엔드와 소통하여 로그인, 회원가입 기능 실습
- 댓글 좋아요 기능 구현 및 코드 리팩토링
- 댓글 관련 기능 구현으로 `className` 중복 최소화 과정 진행
- `Lifecycle` 중 `componentDidMount` 이용한 댓글 작성 기능 구현
- `onClick`을 이용하여 `props.id`를 인자로 받아와 댓글 삭제 기능 구현
- 반복 요소 컴포넌트 분리 후 `map`을 통해 렌더링 완료
- 모든 코드 안에서 `id` 사용 제외
- SCSS 가독성 위한 리팩토링, `margin` 속성 중복 제외
- `early return` 사용을 통한 중괄호의 중첩 줄임
- 변수명만 보고도 내용, 형태 등 유추할 수 있도록 리팩토링

## 기타 질문 및 특이 사항
- 코드 리뷰를 통한 리팩토링을 진행하였습니다. 추후 더 좋은 클론 사이트가 될 수 있도록 리팩토링 진행하겠습니다.

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

orosy committed:
Modify: 초기 셋팅 수정

orosy committed:
💄 Modify: Login, Main page 리액트화에 따른 오류 수정

orosy committed:
✨ Add: Main 페이지 Aside 컴포넌트 추가

orosy committed:
💄 Add: Mission 1, 2 구현

orosy committed:
Merge branch 'main' of https://github.com/wecode-bootcamp-korea/24-React-Westagram-4 into feature/taesookim

orosy committed:
💄 Add: Mission 3 구현

orosy committed:
🎨 Modify: SCSS 가독성 위한 코드 수정

orosy committed:
🔥 Modify: Main 페이지에서 Nav 부분 삭제

orosy committed:
✨ Add: 댓글 버튼 값 유무에 따른 활성화 구현

orosy committed:
🎨 Modify: Misson 4 구현 & Styles 공통 스타일 적용에 따른 수정

orosy committed:
📝 Modify: Fontawesome icon 클래스명 JSX화

orosy committed:
🎨 Modify: Main | 부모 페이지로 단방향 데이터 흐름 구현
- Comment 컴포넌트에 있던 state, event 등의 데이터 Main 페이지 이동
- 각 자식 컴포넌트에 props로 데이터 전달 구현

orosy committed:
✨ Add: Feed 좋아요 버튼 구현 외 리팩토링 진행
 - FeedLike 컴포넌트 생성 후, 좋아요 버튼 구현
 - Main의 자식인 FeedLike 컴포넌트에 props로 데이터 전달 구현
 - Login 페이지 가독성 위한 구조분해 할당 진행
 - CommetOne 컴포넌트 가독성 위한 프로퍼티 이름 변경

orosy committed:
🔥 Modify: CommentList 컴포넌트, 파일명 일치화 & 코드 수정
- CommentList 파일명과 컴포넌트명 일치화시켜 혼란 제거
- 그외 불필요한 코드 수정하고 CommentOne의 구조 변경

orosy committed:
🎨 Modifiy: FeedLike -> BtnLike 컴포넌트명 변경으로 인한 재사용 원활화
- FeedLike 컴포넌트에서 BtnLike로 이름 변경
- BtnLike로 변경하여 컴포넌트 재사용을 통한 댓글 좋아요 기능 구현중

orosy committed:
🎨 Modify: Comment 컴포넌트화
- Feed Article 컴포넌트화 진행
- Mock Data 이용한 렌더링 진행

orosy committed:
🎨 Modify: Feed 컴포넌트화 완료
- FeedList 컴포넌트로 UI 최적화

orosy committed:
✨ Modify: 각 Feed별 Comment Mock Data 변경
- 4개의 Feed 각각의 댓글 Mock Data로 진행

orosy committed:
♻️ Modify: Feed 내 Data 모두 Mock Data 변경
- Like: '좋아요' 누른 사람, 카운트 Mock Data화
- Description: 피드 내용, 올린 사람 Mock Data화
- Update: 피드 업데이트 시간 Mock Data화

dudgus5766 commented on src/pages/taesooKim/Aside/Aside.js:
```diff
@@ -1,11 +1,99 @@
-        안녕하세요?
-      </div>
-    )
+      <aside className="Aside">
```
저는 Aside 파일을 빈파일로 두었는데 생각해보니 오른쪽 프로필 피드를 Aside로 적용하는게 더 좋을 것 같네요!
저도 적용해봐야겠어요 감사합니다🙂

orosy replied:
Aside 주시면 아마 레이아웃이 깨지시겠지만, 영현님은 잘 해내실거라 생각합니다🐸

hjkdw95 commented on src/pages/taesooKim/Login/Login.js:
```diff
@@ -1,11 +1,63 @@
-    return <div>
-      안녕하세요?
-    </div>;
+    const { IdVal, PwVal } = this.state;
+    return (
+      <section className="login">
+        <h1 className="title">Westagram</h1>
+        <div className="login-container">
+          <form id="login-form">
+            <input
+              className="username"
+              type="text"
+              placeholder="전화번호, 사용자 이름 또는 이메일"
+              value={IdVal}
+              onChange={this.handleIdInput}
+            />
+            <input
+              className="password"
+              type="password"
+              placeholder="비밀번호"
+              value={PwVal}
+              onChange={this.handlePwInput}
+            />
```
구조 분해 할당은 `props`로 받아오는 요소에만 사용하는거라고 생각했었는데,  `state`요소에도 직접 사용할 수 있네요! 좋은 사용법 같습니다:)

orosy replied:
구조분해 할당을 많이 하려고 노력했네요🐸

hjkdw95 commented on src/pages/taesooKim/Main/Feed.js:
```diff
@@ -0,0 +1,97 @@
+    this.setState({ isClicked: !this.state.isClicked });
+  };
+  render() {
+    const {
+      name,
+      profile,
+      description,
+      image,
+      value,
+      addComment,
+      getValue,
+      comment,
+      like,
+      update,
+    } = this.props;
```
prettier를 사용하니까 여러 요소를 구조 분해 할당해도 훨씬 눈에 잘들어오네요:) 
아래에 return 부분도 덕분에 보다 직관적으로 눈에 잘들어오는 것 같습니다

orosy replied:
역시 갓갓 prettier

dudgus5766 commented on src/pages/taesooKim/Main/Feed.js:
```diff
@@ -0,0 +1,97 @@
+        </div>
+        <img className="feeds-image" src={image} alt="cookies" />
+        <div className="reaction">
+          <BtnLike
```
`Feed.js`에 직접 state를 두고 하트 버튼 기능 구현할 수도 있겠네요!

orosy replied:
간단하게 적용해보려고 했는데 아직 댓글 좋아요는 구현을 못했네요😢

hjkdw95 commented on src/pages/taesooKim/Main/Feed.js:
```diff
@@ -0,0 +1,97 @@
+              value={value}
+              onChange={getValue}
+            />
+            <button className={`btn-post ${value ? "active" : ""}`}>
```
className 안에서 Javascript 삼항연산자 접근법 좋네요! 👍

dudgus5766 replied:
저도 이건 생각도 못했네요!ㅎㅎ

orosy replied:
저도 구글링한건데 좋은 방법인 것 같아서 구현해봤습니다🐸

hjkdw95 commented on src/pages/taesooKim/Main/FeedList.js:
```diff
@@ -0,0 +1,39 @@
+  render() {
+    const { feedList } = this.props;
+    return (
+      <article className="feeds-container">
+        {feedList.map(feed => {
+          return (
+            <Feed
+              key={feed.id}
+              name={feed.userName}
+              profile={feed.profileImg}
+              description={feed.description}
+              image={feed.feedImg}
+              like={feed.like}
+              comment={feed.comment}
+              update={feed.update}
+            />
+          );
+        })}
+      </article>
```
모두 컴포넌트화 되어있고, 모든 데이터를 json으로 가져오니까 직관적으로 코드가 눈에 너무 잘들어옵니다:)! 정말 보기 좋네요!

orosy replied:
너무 컴포넌트가 많아서 이게 맞는건지는 사실 잘 모르겠지만 감사합니다😁

hjkdw95 commented on src/pages/taesooKim/Main/Like.js:
```diff
@@ -0,0 +1,17 @@
+      <div className="like">
+        <img src={image} alt="orosy" />
+        <p className="many-likes">
+          <span>{name}</span>님 <span>외 {count}명</span>이 좋아합니다
```
`<span>외 {count}명</span>`은 `외 <span>{count}<span>명` 으로 사용해도 괜찮아보이는데, 혹시 이렇게 기재하신 이유가 있으실까요?! 
강조를 위해 사용하는 경우 `strong`태그를 사용해도 좋을 것 같습니다 :)

orosy replied:
사실 저도 최근에 strong을 알게되었는데 이렇게 콕 집어주시다니 증맬 감사해요 현재님 🐸

dudgus5766 commented on src/pages/taesooKim/Main/Feed.js:
```diff
@@ -0,0 +1,97 @@
+    this.setState({ isClicked: !this.state.isClicked });
+  };
+  render() {
+    const {
```
구조 분해 할당..벨로그에 적고 있는데 이렇게 쓰면 좋겠네요!

orosy replied:
구조분해 할당 최고입니다👍

hjkdw95 commented on src/pages/taesooKim/Main/Main.js:
```diff
@@ -1,11 +1,63 @@
-      안녕하세요?
-    </div>;
+    return (
+      <section className="section-main">
+        <Nav />
+        <main className="main">
+          <section className="feeds">
+            <FeedList
+              value={this.state.value}
+              feedList={this.state.feedList}
+              getValue={this.getValue}
+              addComment={this.addComment}
+            />
+          </section>
+          <Aside />
+        </main>
+      </section>
```
완벽한 component화..아름답습니다✨  이렇게 최대한 컴포넌트 화 하여 main에는 state와 각종 function를 집중적으로 보여주는 것이 좋아보입니다.

orosy replied:
좋아보이신다니 감사할 따름이네요😎

hjkdw95 commented on src/pages/taesooKim/Aside/Aside.scss:
```diff
@@ -1 +1,173 @@
+          cursor: pointer;
+        }
+
+        .recomm-namebox {
+          width: 170px;
+          margin: auto 0;
+          font-size: 14px;
+
+          .recomm-username {
+            font-weight: 700;
+            cursor: pointer;
+          }
+
+          .recomm-others {
+            margin-top: 3px;
+            color: var(--color-light-gray);
+          }
+        }
+
+        .recomm-follow {
+          font-size: 14px;
+          color: var(--color-blue);
+          cursor: pointer;
```
태수님 className 작명 정말 멋진거같습니다! 보면 무슨 내용인지 바로 이해가 되네요. 저도 `-`를 보다 적극 활용해서 직관적으로 쓰이도록 노력해야겠습니다:)

orosy replied:
className 작성할 때마다 하나하나 고민했었는데 감사합니당 ✍️

hjkdw95 commented on src/pages/taesooKim/Login/Login.scss:
```diff
@@ -1 +1,78 @@
+/* Global */
+:root {
+  /* Color */
+  --color-white: #ffffff;
+  --color-light-white: #fafafa;
+  --color-dark-white: #dbdbdb;
+  --color-light-blue: #b2dffc;
+  --color-regular-blue: #0095f6;
+  --color-dark-blue: #4f7599;
+}
```
color 변수화 좋습니다👍

orosy replied:
감사합니다👍

dudgus5766 commented on src/pages/taesooKim/Main/Main.js:
```diff
@@ -1,11 +1,63 @@
-    return <div>
-      안녕하세요?
-    </div>;
+    return (
```
`main.js`에서 각 영역 별로 컴포넌트를 나눠놔서  코드 읽기에 훨씬 편하네요.
그런데 혹시 `<section>`을 따로 나눈 이유가 있을까요?

orosy replied:
영현님! 댓글이 "그런데 혹시 을 따로 나눈 이유가 있을까요?"라고 보여서 뭔 말인지 모루겠어여ㅠㅠ

dudgus5766 replied:
`<section>`이었습니다 ㅎㅎ 늦게봐서 죄송해요ㅠ

hjkdw95 reviewed (commented):
태수님 코드 잘봤습니다:) 깔끔한 component 구성과 직관적인 className사용이 너무 인상깊었습니다!

dudgus5766 commented on src/pages/taesooKim/Main/Feed.js:
```diff
@@ -0,0 +1,97 @@
+              value={value}
+              onChange={getValue}
+            />
+            <button className={`btn-post ${value ? "active" : ""}`}>
```
백틱을 사용하니 className 내에서도 바로 함수형이 가능하네요! 간단한 이벤트는 이렇게 적용해도 좋을 것 같아요🙂

orosy replied:
넵 저도 구글링한 덕택에 좋은 코드 하나 찾아냈습니다🐸

dudgus5766 commented on src/pages/taesooKim/Login/Login.scss:
```diff
@@ -1 +1,78 @@
-// login
+/* Global */
+:root {
```
root 사용하니까 정말 편해보이네요... 리팩토링 시에 저도 적용해봐야겠네요
색 뿐만 아니라 폰트에도 적용해도 좋을 것 같네요!

orosy replied:
맞습니다! 폰트에도 적용하려 했는데 그리 사이즈가 많지는 않아서 여기선 적용을 안했네요🐸

dudgus5766 commented on public/data/feedData.json:
```diff
@@ -0,0 +1,146 @@
+[
+  {
+    "id": 1,
```
mock data 양이 어마어마..👍

orosy replied:
mock data와 친해지려고 많이 사용해봤는데.. 아직은 쉽지 않네요🤔

dudgus5766 reviewed (commented):
태수님! 수고하셨습니다🙂 
특히 main.js에서 각 영역으로 컴포넌트를 나누는 건 저도 꼭 적용해봐야겠어요! 
감사합니다!

hjkdw95 commented on src/pages/taesooKim/Login/Login.scss:
```diff
@@ -1 +1,78 @@
+  --color-dark-blue: #4f7599;
+}
+
+.login {
```
저희 컴포넌트 이름을 `loginTaesoo`이런 식으로 줬듯이, 클래스 명을 각 공동 컴포넌트 명으로 정정해야 conflict가 안날 것 같습니다~! 확인 부탁드려요

orosy replied:
감사해요 현재님~ 덕분에 Confilct 방지 해결! 🐸

hjkdw95 reviewed (commented):
추가 comment 확인 부탁드립니다:)

orosy committed:
✏️ Modify: Conflict 방지를 위한 컴포넌트명 변경
- LoginTaeSoo, MainTaesoo로 변경

orosy committed:
♻️ Modify: Login 페이지 유효성 검사 로직 리팩토링
- form 태그로 기능 불가하여 div 태그로 변경
- event target 객체 형태로 활용

orosy committed:
🎨 Modify: 댓글 추가 구현 및 fetch 함수 사용
- Mock Data 사용 후, 달라진 context에 따른 댓글 추가기능 리팩토링
- fetch 함수 사용하여 백엔드와 소통하여 로그인, 회원가입 기능 실습

orosy committed:
✨ Modify: Feed 좋아요 클릭시 하트 색 변경
- 검은 색에서 빨간 색으로 변경

orosy committed:
✨ Add: 댓글 좋아요 기능 구현 및 코드 리팩토링
- 댓글 좋아요 기능 구현
- 댓글 관련 기능 구현으로 className 중복 최소화 과정 진행
- 댓글 삭제 기능 구현중

orosy committed:
♻️: Lifecycle 이용한 댓글 작성 기능 구현
- componentDidMount를 활용하여 댓글 작성 구현
- 이에 더하여 댓글 삭제 구현중

orosy committed:
🎨 Modify: Comment 공백 시 '게시' 불가
- 댓글이 공백이라면, 댓글 게시 불가

orosy committed:
💄 Add: Main 페이지 댓글 삭제 구현
- componentDidMount 활용한 댓글 삭제 기능 구현
- onClick을 이용하여 props id를 통한 기능 구현
- 댓글 삭제 기능 구현중

saengmotmi commented on src/pages/taesooKim/Aside/Aside.js:
```diff
@@ -1,11 +1,99 @@
+                  <p className="story-time">16분 전</p>
+                </div>
+              </li>
+              <li>
```
반복되는 요소들 컴포넌트로 분리한 뒤 반복문 적용해보시면 좋을거 같습니다!

orosy replied:
네네 수정하였습니다🐸

saengmotmi commented on src/pages/taesooKim/Login/Login.js:
```diff
@@ -1,11 +1,86 @@
+import React from "react";
+import { withRouter } from "react-router-dom";
+import "./Login.scss";
+import "../../../styles/common.scss";
```
common.scss는 index.js에서 한번 적용시켜주고 나면 다른 곳에서 굳이 import 할 필요 없을거 같아요

orosy replied:
네네 수정하였습니다🐸

saengmotmi commented on src/pages/taesooKim/Login/Login.js:
```diff
@@ -1,11 +1,86 @@
+    this.state.id.indexOf("@") !== -1 && this.state.pw.length >= 5
+      ? this.setState({ disabled: false })
+      : this.setState({ disabled: true });
+  };
```
동일한 리뷰가 있어서 링크로 대체합니다
https://github.com/wecode-bootcamp-korea/24-React-Westagram-4/pull/2#discussion_r697037352

orosy replied:
네네 수정하였습니다🐸

saengmotmi commented on src/pages/taesooKim/Login/Login.js:
```diff
@@ -1,11 +1,86 @@
+  };
+
+  handleInput = e => {
+    const { value, id } = e.target;
```
```suggestion
    const { value, name } = e.target;
```
id는 앱 전체에 유니크한 셀렉터로서의 역할도 하고 있기 때문에 영향 없는 name이 좀 더 적절한 프로퍼티라고 생각되네요

orosy replied:
네네 수정하였습니다🐸

saengmotmi commented on src/pages/taesooKim/Login/Login.js:
```diff
@@ -1,11 +1,86 @@
+            <button
+              className={`btn ${
+                id.indexOf("@") !== -1 && pw.length >= 5 ? "active" : ""
+              }`}
```
```suggestion
              className={`btn ${ isActive ? "active" : "" }`}
```
로직만 읽어서는 한눈에 무슨 뜻인지 알아보기 힘드니 render 안쪽에 변수로 선언해두고 위와 같이 사용하면 어떨까요?

orosy replied:
네네 수정하였습니다🐸

saengmotmi commented on src/pages/taesooKim/Login/Login.scss:
```diff
@@ -1 +1,79 @@
+    #login-form {
+      input {
+        display: block;
+        width: 300px;
```
```suggestion
  background-color: var(--color-white);
  
  .title {
    display: flex;
    justify-content: center;
    margin: 40px 0 50px;
    font-size: 50px;
    font-family: "Lobster";
  }
  
  .login-container {
    margin-top: 50px;
    
    .login-form {
      input {
        display: block;
        width: 300px;
```
- 새로운 셀렉터 앞에서 한줄 띄우기는 가독성 측면에서 기본적으로 적용해주셨으면 좋겠습니다. 링크 참고해보세요! https://www.notion.so/wecode/Westagram-JS-Refactoring-Checklist-dcff20de07b6449d89d82980e7af9cba#f909425e186044db85a827444dec61a9
- margin 단축 속성 사용가능합니다
- id 셀렉터는 정말 앱 전체에서 무조건 유일무이한 경우가 확실하지 않다면 쓸 필요가 크게 없을 거 같습니다

orosy replied:
네네 수정하였습니다🐸

saengmotmi commented on src/pages/taesooKim/Login/Login.scss:
```diff
@@ -1 +1,79 @@
+        width: 300px;
+        padding: 10px;
+        margin: 0 auto;
+        margin-bottom: 5px;
```
margin을 두번 사용하는게 어색해보여서 하나로 합쳐주시면 좋을거 같아요

orosy replied:
네네 수정하였습니다🐸

saengmotmi commented on src/pages/taesooKim/Login/Login.scss:
```diff
@@ -1 +1,79 @@
+      }
+    }
+  }
+  p {
```
동일한 리뷰라서 링크 참고해주세요
https://github.com/wecode-bootcamp-korea/24-React-Westagram-4/pull/2#discussion_r697035894

orosy replied:
네네 수정하였습니다🐸

saengmotmi commented on src/pages/taesooKim/Main/CommentList.js:
```diff
@@ -0,0 +1,25 @@
+    const { comment, deleteComment } = this.props;
+    return (
+      <div>
+        {comment.map(comment => {
```
```suggestion
        {comments.map(comment => {
```
변수명을 복수형으로 지어주면 변수명만 보고도 배열 형태라는 걸 쉽게 추론할 수 있을듯 하네요

orosy replied:
네네 수정하였습니다🐸

saengmotmi commented on src/pages/taesooKim/Main/Feed.js:
```diff
@@ -0,0 +1,119 @@
+      this.setState({ comment, value: "" });
+    }
+    e.preventDefault();
+  };
```
```suggestion
  handleAdd = e => {
    const { value, comment: prevComments } = this.state
    e.preventDefault();
    
    if (!value) return
    
    const newComment = {
      id: Date.now(),
      userName: "orosy.ts",
      content: value,
      isUser: true,
    };
    
    const comments = [...prevComments, newComment];
    
    this.setState({ comments, value: "" });
  };
```
early return을 적절히 활용하면 중괄호의 중첩을 줄일 수 있습니다

orosy replied:
네네 수정하였습니다🐸

saengmotmi commented on src/pages/taesooKim/Main/Feed.js:
```diff
@@ -0,0 +1,119 @@
+          </div>
+        </div>
+        <div>
+          {like.map(like => {
```
```suggestion
          {likes.map(like => {
```
마찬가지로 복수형 써주시면 좋을거 같아요

orosy replied:
네네 수정하였습니다🐸

saengmotmi commented on src/pages/taesooKim/Main/Feed.js:
```diff
@@ -0,0 +1,119 @@
+  render() {
+    const { value, isClicked, comment } = this.state;
+    const { name, profile, description, image, like, update } = this.props;
+    return (
```
```suggestion
    const { name, profile, description, image, like, update } = this.props;
    
    return (
```
한줄 띄워주면 구분 더 쉬울거 같습니다

orosy replied:
네네 수정하였습니다🐸

saengmotmi reviewed (commented):
고생하셨습니다! 리뷰 확인해주세요~

saengmotmi commented on src/pages/taesooKim/Main/FeedList.js:
```diff
@@ -0,0 +1,40 @@
+              like={feed.like}
+              commentList={feed.commentList}
+              update={feed.update}
+            />
```
데이터 넘겨주실 때 feed로 데이터를 통째로 넘기는 동시에, 각각의 프로퍼티를 분리해서 보내고 있어서 중복인 것 같습니다. 둘 중 하나를 선택해주세요.

orosy replied:
네네 수정하였습니다🐸

orosy committed:
♻️ Modify: 전 컴포넌트 리팩토링 진행
- 반복 요소 컴포넌트 분리 후 map을 통해 렌더링 완료
- 모든 코드 안에서 id 사용 제외
- SCSS 가독성 위한 리팩토링, margin 속성 중복 제외
- early return 사용을 통한 중괄호의 중첩 줄임
- 변수명만 보고도 내용, 형태 등 유추할 수 있도록 리팩토링
